_posts/2022-03-01-auto-backport-in-opensearch.markdown

```diff
@@ -0,0 +1,46 @@
+
+2. The original [backport](https://github.com/tibdex/backport) Github Action created backport branches of the format backport-${pullRequestNumber}-to-${base} which conflicted with other branch protection rules. To maintain branch permissions for such custom branches, we ended up adding a [custom branch name](https://github.com/VachaShah/backport#features) feature on the Github Action which allows passing a branch name prefix as an input, so if the branch name prefix passed is `backport/backport-${pullRequestNumber}`, the workflow will create a branch named `backport/backport-${pullRequestNumber}-to-${base}` for auto-backports.
+ 
+3. For pull requests generated automatically for backports, it is also necessary to have all the required CI workflows running on them similar to a normal pull request. This helps validate the backport pull requests against release branches preventing an incompatible merge. However, according to Github's [documentation](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow), when an action in a workflow is performed using the `GITHUB_TOKEN`, it cannot trigger a new workflow run. This means that when the backport Github Action runs on the original pull request and creates a backport pull request as a result of the workflow, CI workflows cannot be triggered on such pull requests. To solve this problem, we created a simple Github App with required permissions so it can work with the backport workflow to create backport pull requests instead of using the `GITHUB_TOKEN`.
+
+4. The branches created by the backport workflow to open backport pull requests are opened against the repository. Once the pull request is merged, such branches are left over and need to be deleted manually. To resolve this problem and make the process end-to-end automated, we wanted to incorporate a feature where the source branch of the backport pull request is auto deleted after the pull request is merged. This helps to maintain branches efficiently on the repo and allows the backport workflow to clean up after itself. We have added an [auto delete](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/delete_backport_branch.yml) workflow to the release branches which can clean up the auto backport branches after the backport pull requests are merged.
+
+## Current Features
+
+The current state of auto-backport after the learnings is:
```

Comment:
'The current state of auto-backport after the learnings is:' -> 'The current state of auto-backport after these learnings is:'